python/neuroglancer/pipeline/storage.py

```diff
@@ -38,59 +44,76 @@ def __init__(self, layer_path='', n_threads=20):
         self._n_threads = n_threads
 
         if self._path.protocol == 'file':
-            self._interface = FileInterface
+            self._interface_cls = FileInterface
         elif self._path.protocol == 'gs':
-            self._interface = GoogleCloudStorageInterface
+            self._interface_cls = GoogleCloudStorageInterface
         elif self._path.protocol == 's3':
-            self._interface = S3Interface
+            self._interface_cls = S3Interface
+
+        self._interface = self._interface_cls(self._path)
+
+        self._queue = Queue.Queue(maxsize=0) # infinite size
```

Comment:
In theory, if you're handling big files you should be aware of the issues. I honestly don't think this is a big issue either way, so do whatever you're happy with. The downside is that the TaskQueue is no longer completely async.